doc/api/inspector.md

```diff
@@ -108,6 +108,49 @@ with an error. [`session.connect()`] will need to be called to be able to send
 messages again. Reconnected session will lose all inspector state, such as
 enabled agents or configured breakpoints.
 
+## inspector.ws
+
+The Inspector can be interacted with via a websocket-based protocol. These APIs
+allow programatic control of the websocket, equivalent to use `--inspect*` CLI
+options.
+
+### ws.listen()
+
+Start the websocket listener.
+
+### ws.unlisten()
+
```

Comment:
ok. though it sounds like @eugeneo is looking for quite a different API, this is more of a POC at the moment.